src/dapp/walletProvider.ts

```diff
@@ -289,6 +291,84 @@ export class WalletProvider implements IDappProvider {
+     * @param transaction
+     * @param options
+     */
+    async signTransaction(transaction: Transaction, options?: { callbackUrl?: string }): Promise<Transaction> {
+        if (!this.mainFrame) {
+            throw new Error("Wallet provider is not initialised, call init() first");
+        }
+
+        const { contentWindow } = this.mainFrame;
+        if (!contentWindow) {
+            throw new Error("Wallet provider is not initialised, call init() first");
+        }
+
+        return new Promise<Transaction>((resolve, reject) => {
+            let plainTransaction = WalletProvider.prepareWalletTransaction(transaction);
+            console.log("postMessage", DAPP_MESSAGE_SIGN_TRANSACTION_URL, plainTransaction);
+
+            contentWindow.postMessage({
+                type: DAPP_MESSAGE_SIGN_TRANSACTION_URL,
+                data: {
+                    transaction: plainTransaction
+                }
+            }, this.walletUrl);
+
+            const timeout = setTimeout(_ => reject('sign transaction url not responding'), 5000);
+            const signTransactionUrl = (ev: IDappMessageEvent) => {
+                console.log("event", "signTransactionUrl", ev);
+
+                if (!this.isValidWalletSource(ev.origin)) {
+                    return;
+                }
+
+                const { data } = ev;
+                if (data.type !== DAPP_MESSAGE_SIGN_TRANSACTION_URL) {
+                    return;
+                }
+
+                clearTimeout(timeout);
+                window.removeEventListener('message', signTransactionUrl.bind(this));
+
+                if (data.error) {
+                    return reject(data.error);
+                }
+
+                return resolve(data.data.toString());
+            };
+
+            window.addEventListener('message', signTransactionUrl);
+        }).then((url: any) => {
+            window.location.href = `${this.baseWalletUrl()}${url}&callbackUrl=${options !== undefined && options.callbackUrl !== undefined ? options.callbackUrl : window.location.href}`;
+            return transaction;
+        });
+    }
+
+    /**
+     * Method will be available once the ElrondWallet hook will be implemented
+     * @param _
+     */
+    async signMessage(_: SignableMessage): Promise<SignableMessage> {
+        throw new ErrNotImplemented();
+    }
+
+    static prepareWalletTransaction(transaction: Transaction): any {
+        let plainTransaction = transaction.toPlainObject();
+
+        // We adjust the fields, in order to make them compatible with what the wallet expected
+        plainTransaction["nonce"] = transaction.getNonce().valueOf();
+        plainTransaction["data"] = transaction.getData().valueOf().toString();
+        plainTransaction["value"] = transaction.getValue().toString();
+        plainTransaction["gasPrice"] = transaction.getGasPrice().valueOf();
+        plainTransaction["gasLimit"] = transaction.getGasLimit().valueOf();
+
```

Comment:
Chain ID and transaction spec version?

Reply:
Same as above - these were changes from the development branch.